extensions/wikia/Scribe/ScribeEventProducerController.class.php

```diff
@@ -1,30 +1,22 @@
 <?php
 
-class ScribeEventProducerController {
+use Wikia\Logger\WikiaLogger;
 
-	/**
-	 * Ugly fix! TODO
-	 * The following static properties are declared to store
-	 * $oPage, $oUser and $oRevision objects passed from SaveComplete
-	 * and SaveRevisionComplete hooks. They are necessary to call
-	 * the buildEditPackage() method. For files, to check if they are local,
-	 * we need to wait till the UploadComplete hook is run.
-	 * This hook only passes an instance of UploadBase class so it is needed
-	 * to store the mentioned objects from the previously run hooks.
-	 * It is going to be fixed with the new version of the ImageReview tool.
-	 * @see    CE-1125 or any ImageReview related ticket
-	 * @blame  Adam Karminski <adamk@wikia-inc.com>
-	 */
-	private static $oPage, $oUser, $oRevision;
+class ScribeEventProducerController {
 
 	static public function onUploadComplete( UploadBase $oForm ) {
 		wfProfileIn( __METHOD__ );
 
-		$oLocalFile = $oForm->getLocalFile();
+		$title = $oForm->getTitle();
+		if ( $title instanceof Title ) {
```

Comment:
Maybe check for `if ($title && $title->exists()) {` would suite you better here?